execution/limits/limits.go

```diff
@@ -0,0 +1,32 @@
+package limits
+
+import (
+	"sync"
+
+	"github.com/efficientgo/core/errors"
+	"go.uber.org/atomic"
+)
+
+type Limits struct {
+	maxSamples int
+
+	curSamplesPerTimestamp sync.Map
+}
+
+func NewLimits(maxSamples int) *Limits {
+	return &Limits{
+		maxSamples: maxSamples,
+	}
+}
+
+func (l *Limits) AccountSamplesForTimestamp(t int64, n int) error {
+	v, _ := l.curSamplesPerTimestamp.LoadOrStore(t, atomic.NewInt64(0))
```

Comment:
If we would do: `if l.maxSamples == 0 { return nil }` then the zero value of `Limits` would also be useful and we wouldn't have to do the previous logic.

Reply:
done